engine/test-src/org/pentaho/di/trans/steps/http/HTTPIT.java

```diff
@@ -148,10 +194,38 @@ public void test204Answer() throws Exception {
     when( stepMockHelper.processRowsStepMetaInterface.getFieldName() ).thenReturn( "ResultFieldName" );
     http.init( stepMockHelper.processRowsStepMetaInterface, data );
     assertTrue( http.processRow( stepMockHelper.processRowsStepMetaInterface, data ) );
-    System.out.println( Arrays.toString( expectedRow ) );
     Object[] out = ( (HTTPHandler) http ).getOutputRow();
-    System.out.println( Arrays.toString( out ) );
-    assertTrue( Arrays.equals( expectedRow, out ) );
+    assertTrue( meta.equals( out, expectedRow, index ) );
+  }
+
+  @Test
+  public void testResponseHeader() throws Exception {
+    HTTPData data = new HTTPData();
+    int[] index = { 0, 1, 2 };
+    RowMeta meta = new RowMeta();
+    meta.addValueMeta( new ValueMetaString( "fieldName" ) );
+    meta.addValueMeta( new ValueMetaInteger( "codeFieldName" ) );
+    meta.addValueMeta( new ValueMetaString( "headerFieldName" ) );
+    Object[] expectedRow =
+      new Object[] { "", 402L, "{\"host\":\"localhost\"}" };
+    HTTP http =
+      new HTTPHandler( stepMockHelper.stepMeta, data, 0, stepMockHelper.transMeta, stepMockHelper.trans, true );
+    RowMetaInterface inputRowMeta = mock( RowMetaInterface.class );
+    http.setInputRowMeta( inputRowMeta );
+    when( inputRowMeta.clone() ).thenReturn( inputRowMeta );
+    when( stepMockHelper.processRowsStepMetaInterface.getUrl() ).thenReturn( HTTP_LOCALHOST_9998 );
+    when( stepMockHelper.processRowsStepMetaInterface.getHeaderField() ).thenReturn( new String[] {} );
+    when( stepMockHelper.processRowsStepMetaInterface.getArgumentField() ).thenReturn( new String[] {} );
+    when( stepMockHelper.processRowsStepMetaInterface.getResultCodeFieldName() ).thenReturn( "ResultCodeFieldName" );
+    when( stepMockHelper.processRowsStepMetaInterface.getFieldName() ).thenReturn( "ResultFieldName" );
+    when( stepMockHelper.processRowsStepMetaInterface.getEncoding() ).thenReturn( "UTF8" );
+    when( stepMockHelper.processRowsStepMetaInterface.getResponseHeaderFieldName() ).thenReturn(
+      "ResponseHeaderFieldName" );
```

Comment:
@eliofreitas
you had already told me this, and i replied that wasn't what i meant. next time you don't understand an answer it's better to ask before updating the pull request instead of reiterating the question in it
but it was a very minor detail and really not important, let it be